src/cmd/api/controllers/video_upload_handler.go

```diff
@@ -81,14 +80,45 @@ func (v VideoUploadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// Create new video
 	videoCreated, err := dao.CreateVideo(v.MariadbClient, videoID, title, int(models.UPLOADING))
 	if err != nil {
-		log.Error("Cannot insert new video into database: ", err)
-		w.WriteHeader(http.StatusBadRequest)
-		return
+		// Check if the returned error comes from duplicate title
+		videoCreated, err = dao.GetVideoFromTitle(v.MariadbClient, title)
+		if err != nil {
+			log.Error("Cannot find video ", title, "  : ", err)
+			log.Error("Cannot insert new video into database: ", err)
```

Comment:
2 logs with contradictory messages ?

Reply:
No, we are in case in which video creation failed (ie. insert into database failed). 
Then, we are checking if the insertion failed because the given title already exists (GetVideoFromTitle).
If so, we have to check video status, may be last update/encode failed and user wants to retry.
Or there is another error (database no more reachable for example) and then we "Cannot find video..." and so the new video creation/insertion failed too.